fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingScrubDangling.java

```diff
@@ -72,15 +72,18 @@
     @Nonnull private static final IndexBuildProto.IndexBuildIndexingStamp myIndexingTypeStamp = compileIndexingTypeStamp();
 
     @Nonnull private final OnlineIndexScrubber.ScrubbingPolicy scrubbingPolicy;
+    @Nonnull private final AtomicLong danglingCount;
     private long scanCounter = 0;
     private int logWarningCounter;
 
     public IndexingScrubDangling(@Nonnull final IndexingCommon common,
                                  @Nonnull final OnlineIndexer.IndexingPolicy policy,
-                                 @Nonnull final OnlineIndexScrubber.ScrubbingPolicy scrubbingPolicy) {
+                                 @Nonnull final OnlineIndexScrubber.ScrubbingPolicy scrubbingPolicy,
+                                 @Nonnull final AtomicLong danglingCount) {
```

Comment:
What's the use case for supplying the dangling count as a constructor parameter instead of initializing the `danglingCount` to a `new AtomicLong(0)`? It looks like all the methods I could see always passed in 0, unless you're anticipating something like invoking the scrubber multiple times and pre-seeding it with some non-zero count

Reply:
The `AtomicLong` object is defined within the `scrubDanglingIndexEntries/scrubMissingIndexEntries` functions and, IIUC, does not accumulate between calls. The reason it is defined in the object's constructor and not in the scrubbing function is that the general purpose indexing/scrubbing function `buildIndexAsync`, which is implemented by `indexingBase`, has no knowledge of this scrubber-specific counter.

Reply:
Why not move the instantiation from the `scrubFooIdexEntries` methods to the constructors (and remove the parameters entirely)?

Reply:
How should it transfer the value back to the `scrubFooIndexEntries` methods?

Reply:
Oh, I see. You're relying on that value being updated as a side effect of the scrubbing. I'd missed that and assumed it was now part of the scrub return signature.

That feels like something we'll want to clean up in the future (and allow the scrubbing methods to return a more interesting response object--you could imagine getting a list of index violations, for example, perhaps truncated to just the first few), but maybe it's okay for now